versioned_docs/version-4.0/devupdate.md

```diff
@@ -0,0 +1,958 @@
+
+:::
+
+##### Theme updates
+
+With the changes in boost to incorporate the primary and secondary navigation, third party themes would need to account for the following in their templates:
+
+- To leverage the activity_header, it's data needs to be exported and included into the base template :
+  - `headercontent` being the array element that contains the exported activity_header data
+
+  ```html
+  {{#headercontent}}
+     {{> core/activity_header}}
+  {{#headercontent}}
+
+   ```
+
+- It is recommended to transition towards the secondary/tertiary navigation hierarchy to reduce user cognitive load and with a logical separation of components
+  - Secondary navigation can be added to the templates by following the example https://github.com/moodle/moodle/blob/master/theme/boost/templates/columns2.mustache#L64-L68 This leverages the secondary navigation class to generate it's content.
+- Flat navigation classes have been marked for deprecation. Themes that leverage the flat_navigation will need to make the following changes in their plugins in order to use it
+  - Account for the additional changes [theme changes](#theme-changes)
+  - Indicate that they do not implement secondary navigation via the page's `set_secondary_navigation` function. It is recommended to set this within the root layout file, for example, columns2
+  - Initialise the flat navigation by introducing the following in the root layout file (if not existent)
+
+   ```php
+   $nav = $PAGE->flatnav;
+   $templatecontext['firstcollectionlabel'] = $nav->get_collectionlabel();
+   ```
+
+- In order to reintroduce the settings cog in the templates, you can introduce the following:
+
+  ```html
+  <div id="region-main-settings-menu" class="d-print-none {{#hasblocks}}has-blocks{{/hasblocks}}">
+     <div> {{{ output.region_main_settings_menu }}} </div>
+  </div>
+  ```
+
+:::info Accessibility notes
+
+The jump to `maincontent` div is now rendered within the activity header when within an activity context.
+
+:::
+
+## Component library
+
+Each Moodle installation now ships with a Moodle User Interface (UI) Component library, a documentation system used to describe all the Bootstrap components and the custom Moodle components. The component Library is a helper tool for developers when creating user interfaces, a testing tool for theme developers and a documentation tool for core developers. The ultimate goal of having a component library is to encourage developers to create consistent user interfaces to improve Moodle's overall user experience.
+
+The library contains pages with documentation about User Interface components. It contains details on how to use the component, what variations are available and the JavaScript events / options are associated with the component.
+
+When writing on these pages it is possible to render core mustache templates using some custom syntax like this:
+
+```mustache
+ {{< mustache template="core/notification_error" >}}
+ {{< /mustache >}}
+```
```

Comment:
```suggestion
```mustache
{{< mustache template="core/notification_error" >}}
{{< /mustache >}}
```
```

Reply:
I don't think these blockquotes should be closed...